src/Plutarch/Extra/AssetClass.hs

```diff
@@ -392,3 +399,18 @@ instance
   LabelOptic "name" k (Tagged tag AssetClass) (Tagged tag' AssetClass) a b
   where
   labelOptic = #unTagged %% #name
+
+----------------------------------------
+-- Ply instances
+
+-- | @since 3.10.3
+instance PlyArg AssetClass where
```

Comment:
```hs
instance PlyArg AssetClass where
  type UPLCRep AssetClass = [Plutus.Data]
  toBuiltinArg ac =
    case Plutus.toData @AssetClass ac of
      Plutus.List x -> x
      _ -> error "unreachable"
  toBuiltinArgData = Plutus.toData
```

Sorry, I should've explained it better. Basically, we don't want to expose actual logic here, so that when we change `AssetClass`, it doesn't break.

Reply:
How about using `PlyArg` as the single source of truth so that we don't have to do deconstruction?

Reply:
the single source of truth should be the derivation result as we don't have control over that. I'm not sure where deconstruction came from 🤔. 

Also, most `PlyArg` instances in ply themselves are defined similar to this.

Reply:
Good point.

Reply:
Thanks!